Content.Shared/Pinpointer/PinpointerComponent.cs

```diff
@@ -11,52 +12,89 @@ namespace Content.Shared.Pinpointer;
 [Access(typeof(SharedPinpointerSystem))]
 public sealed partial class PinpointerComponent : Component
 {
-    // TODO: Type serializer oh god
-    [DataField("component"), ViewVariables(VVAccess.ReadWrite)]
-    public string? Component;
+    /// <summary>
+    ///     A list of components that will be searched for when selected from the verb menu.
+    ///     The closest entities found with one of the components in this list will be added to the StoredTargets.
+    /// </summary>
+    [DataField]
+    public ComponentRegistry Components = new();
+
+    /// <summary>
+    ///     A list of entities that are stored on the pinpointer
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
+    public readonly List<EntityUid> StoredTargets = new();
 
```

Comment:
Have you tested this with PVS because I doubt this would work as 99% of the time it's going to resolve to invalid entityuids.

Also why are there now 2 target fields not one? That just seems like unnecessary code additions.

Reply:
One of the fields(Components) keeps track of the components the pinpointer is allowed to search for, this will search for the closest target that has the specific component. The EntityUid of this component is then added to the other field(StoredTargets) so it can be searched for even if it's not the closest one with the component anymore.
The EntityUid list also allows storing EntitUid's of targets that were not aquired through a component search.